### PR TITLE
PLATFORM-3548: remove unnecessary path manipulations when doing HTTPS…

### PR DESCRIPTION
**Pull request by harnash**
… downgrades

This affects Sitemaps and robots.txt
related to 
* https://github.com/Wikia/wikia-vcl/pull/1832
* https://github.com/Wikia/wikia-vcl/pull/1833

### CODE REVIEW

**Review by Grunny — APPROVED** [review]

File: extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php [hunk 1]
```diff
@@ -63,36 +63,17 @@ public static function onAfterInitialize( Title $title, $article, OutputPage $ou
...
-	 * @param  WebRequest $request
-	 * @param  User       $user
-	 * @return boolean
-	 */
-	public static function onWikiaRobotsBeforeOutput( WebRequest $request, User $user ): bool {
-		if ( WebRequest::detectProtocol() === 'https' &&
-			!self::httpsAllowed( $user, $request->getFullRequestURL() )
-		) {
-			self::downgradeRedirectForPath( '/robots.txt', $request );
+			self::downgradeRedirectForPath( $request->getFullRequestURL(), $request );
```
Comment (themech) [inline extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php]: For the downgrade, do we need to pass both, the path and the request? Maybe we can just pass the `$request` and rename the method to something like `downgradeRedirect` or `downgradeRequest`.

Comment (harnash) [reply extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php]: No. Currently we don't. 0b8e14a5902

File: extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php [hunk 2]
```diff
@@ -66,27 +66,20 @@ public static function onAfterInitialize( Title $title, $article, OutputPage $ou
...
-	 * Handle downgrading anonymous requests for robots.txt.
-	 *
-	 * @param  WebRequest $request
-	 * @param  User       $user
-	 * @return boolean
-	 */
-	public static function onWikiaRobotsBeforeOutput( WebRequest $request, User $user ): bool {
-		return self::handleRedirectForPath( '/robots.txt', $request, $user );
+	public static function onSitemapRobotsPageBeforeOutput( WebRequest $request, User $user ): bool {
+		if ( WebRequest::detectProtocol() === 'https' &&
```
Comment (Grunny) [inline extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php]: This is no longer just downgrading after https://wikia-inc.atlassian.net/browse/PLATFORM-3552 which is why the method below was renamed to just `handleRedirectForPath`, so having this check here undoes the work in PLATFORM-3552 for robots and sitemaps. I think this hook should now just be the content of what was the `handleRedirectForPath` method below.

**Review by Grunny — CHANGES_REQUESTED** [review]

File: extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php (hunk 2 above)
Comment (harnash) [reply extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php]: Oops. Fixed that in 017ce92037c

**Review by themech — APPROVED** [review]

**Review by Grunny — APPROVED** [review]

**Pull request merged**